alvr/graphics/src/stream.rs

```diff
@@ -287,13 +300,89 @@ impl StreamRenderer {
                 &(view_idx as u32).to_le_bytes(),
             );
             render_pass.set_bind_group(0, &self.views_objects[view_idx].bind_group, &[]);
+            set_passthrough_push_constants(&mut render_pass, passthrough);
             render_pass.draw(0..4, 0..1);
         }
 
         self.context.queue.submit(iter::once(encoder.finish()));
     }
 }
 
+fn set_passthrough_push_constants(render_pass: &mut RenderPass, config: Option<&PassthroughMode>) {
+    const DEG_TO_NORM: f32 = 1. / 360.;
+
+    alvr_common::error!("{:?}", config);
+
+    fn set_float(render_pass: &mut RenderPass, offset: u32, value: f32) {
+        render_pass.set_push_constants(ShaderStages::VERTEX_FRAGMENT, offset, &value.to_le_bytes());
+    }
+
+    match config {
+        Some(PassthroughMode::AugmentedReality { brightness }) => {
+            set_float(render_pass, ALPHA_CONST_OFFSET, 1. - brightness);
+            set_float(render_pass, ENABLE_CHROMA_KEY_CONST_OFFSET, 0.);
+        }
+        Some(PassthroughMode::Blend { opacity }) => {
+            set_float(render_pass, ALPHA_CONST_OFFSET, 1. - opacity);
+            set_float(render_pass, ENABLE_CHROMA_KEY_CONST_OFFSET, 0.);
+        }
+        Some(PassthroughMode::ChromaKey(config)) => {
+            render_pass.set_push_constants(
+                ShaderStages::VERTEX_FRAGMENT,
+                ENABLE_CHROMA_KEY_CONST_OFFSET,
+                &1_u32.to_le_bytes(),
+            );
+
+            set_float(
+                render_pass,
+                CK_TARGET_CONST_OFFSET,
+                config.hue_deg * DEG_TO_NORM,
+            );
+            set_float(
+                render_pass,
+                CK_TARGET_CONST_OFFSET + FLOAT_SIZE,
+                config.saturation,
+            );
+            set_float(
+                render_pass,
+                CK_TARGET_CONST_OFFSET + 2 * FLOAT_SIZE,
+                config.value,
+            );
+
+            set_float(
+                render_pass,
+                CK_WEIGHT_CONST_OFFSET,
+                1. / (config.hue_range_deg * DEG_TO_NORM),
+            );
+            set_float(
+                render_pass,
+                CK_WEIGHT_CONST_OFFSET + FLOAT_SIZE,
+                1. / config.saturation_range,
+            );
+            set_float(
+                render_pass,
+                CK_WEIGHT_CONST_OFFSET + 2 * FLOAT_SIZE,
+                1. / config.value_range,
+            );
+
+            set_float(
```

Comment:
Just split this into a min/max and it'll work way better tbh (a range greater than 1.0 doesn't make a ton of sense anyhow even though I guess the distances can get higher than 1.0)